recipes-extended/xen/files/fix-nesting-2.patch

```diff
@@ -0,0 +1,151 @@
+Index: xen-4.6.4/xen/arch/x86/mm/p2m.c
+===================================================================
+--- xen-4.6.4.orig/xen/arch/x86/mm/p2m.c
++++ xen-4.6.4/xen/arch/x86/mm/p2m.c
+@@ -2024,6 +2024,128 @@ p2m_get_p2m(struct vcpu *v)
+     return p2m_get_nestedp2m(v, nhvm_vcpu_p2m_base(v));
+ }
+ 
++
++
++static unsigned long paging_get_l1_pfn_from_l2_pa(struct vcpu *v, unsigned long l2_pa)
++{
++       unsigned long l1_pfn;
++
++       unsigned int page_order;
++       uint8_t ept_p2m_acc;
++       uint64_t exit_qual;
++       uint32_t exit_reason;
++
++       /* BUG: We use access 0 here - but EPT requires we check the RWX/US from pfec*/
++
++               if (nept_translate_l2ga(v,l2_pa & PAGE_MASK,
++               &page_order, 0, &l1_pfn, &ept_p2m_acc, 
++               &exit_qual, &exit_reason) != EPT_TRANSLATE_SUCCEED) 
++                       return INVALID_GFN;
++
++
++       return l1_pfn;
++}
++
++
++static int paging_read_l2_entry(struct vcpu *v, unsigned long l2_pa, uint64_t * entry)
```

Comment:
Please be consistent with spacing and honor spacing conventions of existing code you are patching.